### DISCUSSION TITLE
Question: Creating a theme that requires a custom plugins / files / configuration

### DISCUSSION THREAD

**Discussion by vbharadwaj-bk** [getpelican/pelican#3252] [Q&A] [answered]
I recently wrote a port of the template [al-folio](https://github.com/alshedivat/al-folio) for Pelican. The original is a popular academic website template written for Jekyll with Liquid templates (MIT license).  The Pelican version is here:<https://github.com/vbharadwaj-bk/al-folio-python>, and you can see a live version on [my website](https://vivek-bharadwaj.com). It might be useful to make this a Pelican theme, since I am sure there are others who would like to use this template with Python instead of Ruby.

My question: this template requires some special pages and content files (e.g. config.yml for a site-wide configuration, homepage has some specific metadata about the profile picture, etc.). Some custom options need to be set in `pelicanconf.py`, and my repo bundles a Pelican plugin that handles some logic on the Python side.

The themes in the Pelican-themes repository currently only seem to have static files and a list of templates, without any "starter code" about what the content directory or settings file should look like. Does this imply that having plugins / additional settings / special pages falls outside the scope of a theme? Or is there some way to include all this "sample content" in the theme directory?

**Comment by avaris** [comment, marked as answer]
> Does this imply that having plugins / additional settings / special pages falls outside the scope of a theme?

Currently... Yes: Themes have no way to "enforce" these conditions, they are merely a bundle of template and static content. And no: themes may have dependencies/requirements/etc but this is communicated via README (at best).

We want to move toward themes as python packages. This should simplify a lot of this. See #1564 for discussion and possible plans for implementation. I am looking for input from theme developers for what the interface and functionality should look like. Feel free to chime in :).

**Reply by vbharadwaj-bk** [reply]
Themes as Python packages seems like a good solution! I will follow the discussion.
